tests/pkcs12.js

```diff
@@ -0,0 +1,57 @@
+const caCert = Buffer.from(
+'308204FF308203E7A003020102021433441AA86C01ECF660F270510A4CD114FABCE944300D06092A864886F70D01010B0500308194310B30090603550406130255533110300E06035504080C074D6F6E74616E613110300E06035504070C07426F7A656D616E3111300F060355040A0C08536177746F6F746831133011060355040B0C0A436F6E73756C74696E673118301606035504030C0F7777772E776F6C6673736C2E636F6D311F301D06092A864886F70D0109011610696E666F40776F6C6673736C2E636F6D301E170D3233313231333232313932385A170D3236303930383232313932385A308194310B30090603550406130255533110300E06035504080C074D6F6E74616E613110300E06035504070C07426F7A656D616E3111300F060355040A0C08536177746F6F746831133011060355040B0C0A436F6E73756C74696E673118301606035504030C0F7777772E776F6C6673736C2E636F6D311F301D06092A864886F70D0109011610696E666F40776F6C6673736C2E636F6D30820122300D06092A864886F70D01010105000382010F003082010A0282010100BF0CCA2D14B21E84425BCD381F4AF24D7510F1B6359FDFCA7D0398D3ACDE0366EE2AF1D8B07D6E07540B1098214D80CB1220E7CC4FDE457DC9727732EACA90BB695210032FA8F395C5F18B62561BEF676FA4104195AD0A9BE3A5C0B0D2707650305BA8E8082C7CEDA7A27A8D38291CACC7EDF27C95B095827D495C38CD7725EFBD807553943C3DCA635B9F15B5D31D132F19D13CDB763ACCB87DC9E5C2D7DA406FD821DC731B422D539CFE1AFC7DAB7A363F98DE847C0567CE6A143887A9F18CB568CB687F71202BF5A063F5562FA326D2B76FB15A17D7389908FE93586FFEC3134908160BA74D6700523167234E98ED51451DB904D90BECD828B34BBDED36790203010001A382014530820141301D0603551D0E04160414278E671174C3261D3FED3363B3A4D81D30E5E8D53081D40603551D230481CC3081C98014278E671174C3261D3FED3363B3A4D81D30E5E8D5A1819AA48197308194310B30090603550406130255533110300E06035504080C074D6F6E74616E613110300E06035504070C07426F7A656D616E3111300F060355040A0C08536177746F6F746831133011060355040B0C0A436F6E73756C74696E673118301606035504030C0F7777772E776F6C6673736C2E636F6D311F301D06092A864886F70D0109011610696E666F40776F6C6673736C2E636F6D821433441AA86C01ECF660F270510A4CD114FABCE944300C0603551D13040530030101FF301C0603551D1104153013820B6578616D706C652E636F6D87047F000001301D0603551D250416301406082B0601050507030106082B06010505070302300D06092A864886F70D01010B050003820101002DFCF9325ABED69D428B864E6722C3502DCB14271D94F3CD8842DA411C392467A7924D27EA568219BF11B243A48D5D87B22764668281DFC4FD5B62B0C24D9D29F24132CC2EB5DA38061BE87F8C6E3D801E005649BF39E0DA682FC4FD00E6D1811AD14ABB7652CE4D249DC4A3A7F165142F1FA82DC6CBCEB1A789742627C3F3A3844C340114037D163AC88B252E7B90CC46B15234BA936EEFFE43A3ADC66F51FBBAEA38E36FD6EE636236EA5E08B4E22A4689E3AEB3B406EF637A6E5DDDC9EC024FF764C02707B46F4A18725B34747CD0A9048F408B6A39D26B1A01F201A881343AE5B055D13C95CAB082D6ED982815597E95A769C7B57BEC01A74DE6B9A2FE35', 'hex');
+
+const pkcs12_tests =
+{
+    createAndParse: function()
+    {
+        let pkcs12Create = new WolfSSL_PKCS12()
+        let pkcs12Parse = new WolfSSL_PKCS12()
+
+        pkcs12Create.Create('my secure password :D', key, cert, [caCert],
```

Comment:
FIPS technically does not allow DES3 and SHA1 in this case. Does PKCS12 support any other algos?

Reply:
it runs and passes with fips-ready, are they supposed to be compiled out? it looks like sha1 is the only option for this step in PKCS12_create_key_content but it later uses sha256 as the default for the digest:
```
static byte* PKCS12_create_key_content(WC_PKCS12* pkcs12, int nidKey,
        word32* keyCiSz, WC_RNG* rng, char* pass, word32 passSz,
        byte* key, word32 keySz, int iter)
{
    byte*  keyBuf;
    word32 keyBufSz = 0;
    byte* keyCi = NULL;
    word32 tmpSz;
    int ret;
    int algo;
    void* heap;

    heap = wc_PKCS12_GetHeap(pkcs12);
    *keyCiSz = 0;
    switch (nidKey) {
        case PBE_SHA1_RC4_128:
            algo = 1;
            break;

        case PBE_SHA1_DES:
            algo = 2;
            break;

        case PBE_SHA1_DES3:
            algo = 3;
            break;

        /* no encryption */
        case -1:
            algo = -1;
            break;

        default:
            WOLFSSL_MSG("Unknown/Unsupported key encryption");
            return NULL;
    }
```

```
WC_PKCS12* wc_PKCS12_create(char* pass, word32 passSz, char* name,
        byte* key, word32 keySz, byte* cert, word32 certSz, WC_DerCertList* ca,
        int nidKey, int nidCert, int iter, int macIter, int keyType, void* heap)
...
        #ifndef NO_SHA256
            mac->oid = SHA256h;
        #elif !defined(NO_SHA)
            mac->oid = SHA;
        #elif defined(WOLFSSL_SHA384)
            mac->oid = SHA384;
        #elif defined(WOLFSSL_SHA512)
            mac->oid = SHA512;
        #else
            WOLFSSL_MSG("No supported hash algorithm compiled in!");
            wc_PKCS12_free(pkcs12);
            wc_FreeRng(&rng);
            return NULL;
        #endif
```

Reply:
@JacobBarthelmeh added support for AES to replace 3DES. For FIPS users we should prefer FIPS algorithms (3DES was forever removed from the FIPS program by NIST on 1 Jan 2024, it is no longer a FIPS algorithm. It may work but it will be outside the module boundary and no longer certifiable. Users with strict FIPS requirements will not be able to use this solution).

SHA1 is still allowed to be used up till Dec 31 2030 however NIST is already encouraging the industry to start abandoning it as we transition into the post quantum era. SHA384, SHA512, SHA512-224 and SHA512-256 are preferred by NSA!